File: src/main/java/org/app/infrastructure/endpoints/dto/GetFanListDTO.java
package org.app.infrastructure.endpoints.dto;

import org.app.domain.User;

import java.util.List;

public class GetFanListDTO {
    public String id;
    public String username;

    public GetFanListDTO(User user) {
        this.id = user.getId().toString();
        this.username = user.getUsername();
    }

    public static List<GetFanListDTO> from(User user){
        return user.getFans().stream().map(GetFanListDTO::new).toList();
    }
}

File: go/app/main/infrastructure/endpoints/dto/get_fan_list_dto.go
package dto

import (
	"mutualAttentionSystem/app/main/domain/events"
	"mutualAttentionSystem/app/main/infrastructure/utils"
)

type GetFanListDTO struct {
    Fans []*GetUserItemDTO
    Page       int
    Limit      int
    TotalPage  int
}

func ToGetFanListDTO(event *events.GetUserEvent, page int, limit int) *GetFanListDTO {
    pageable := utils.NewPableable(page, limit, len(event.User.GetFollowings()))
    
    getFanListDTO := GetFanListDTO{make([]*GetUserItemDTO, 0), page, limit, pageable.TotalPage}
    for _, user := range event.User.GetFans() {
        getFanListDTO.Fans = append(getFanListDTO.Fans, ToGetUserItem(user))
    }
    return &getFanListDTO
}
